### ISSUE TITLE
Box Plot X Axis Labels

### ISSUE THREAD

**Issue by djchou** [pydata/pandas#10034]
It would be nice if there was an option to have box plot X axis labels match line plot's X axis labels.  Specifically, when plotting a time series, box plot labels each tick whereas line plot's labels are more spread apart.

**Comment by TomAugspurger** [comment]
Right now the Timeseries line plots are handled separately. I'm still working on unifying it with the rest of the plotting code so that barplots, scatter plots, etc can use it too.

I don't think this would ever be the default behavior for boxplot though. Box plots are more likely draw for a fixed set of categories.

**Comment by djchou** [comment]
Yeah, I'm not looking to change the default behavior but rather have an argument that allows the label to be toggled.

**Comment by MarcoGorelli** [comment]
closing as there's been no interest in this in nearly a decade

if someone wants this feature, please open a new issue with:
- reproducible example
- expected output
